### DISCUSSION TITLE
Have guys considered using the Martinez algorithm over the current bsp method for boolean operations in V3?

### DISCUSSION THREAD

**Discussion by clupnyluke** [jscad/OpenJSCAD.org#1157] [Ideas]
This algorithm was discovered in 2009 by Francisco Martinez, Antonio Jesus Rueda, Francisco Ramon Feito and operates in O((n + k) log n) time ( n is number of edges and k - is the number of intersections). I am not aware if this is compatible with the way that the current CSG works or not but I thought I would bring it up. [Here is a repo with an implementation of the algorithm](https://github.com/mfogel/polygon-clipping/actions?query=workflow%3Abuild). [Here is the paper in question](https://github.com/mfogel/polygon-clipping/blob/HEAD/paper.pdf)

**Comment by platypii** [comment]
There has been discussions of it. In general, for the 2D boolean operations, Martinez does seem to be the best thing out there. It would be nice to use in jscad.
https://github.com/jscad/OpenJSCAD.org/issues/1120

**Reply by platypii** [reply]
Merged into V3 branch #1177
